sympy/solvers/simplex.py

```diff
@@ -0,0 +1,966 @@
+    >>> A, B, C, D = [Matrix(i) for i in l]
+    >>> _simplex(A, B, -C, -D)
+    (-6, [3, 2], [1, 0, 0, 0])
+    >>> _simplex(A, B, -C, -D, dual=True)  # [5, 0] != [3, 2]
+    (-6, [1, 0, 0, 0], [5, 0])
+
+    In both cases the function has the same value:
+
+    >>> Matrix(C)*Matrix([3, 2]) == Matrix(C)*Matrix([5, 0])
+    True
+
+    See Also
+    ========
+    _lp - poses min/max problem in form compatible with _simplex
+    lpmin - minimization which calls _lp
+    lpmax - maximimzation which calls _lp
+
+    References
+    ==========
+
+    .. [1] Thomas S. Ferguson, LINEAR PROGRAMMING: A Concise Introduction
+           web.tecnico.ulisboa.pt/mcasquilho/acad/or/ftp/FergusonUCLA_lp_args.pdf
+    """
+
+    A, B, C, D = [Matrix(i) for i in (A, B, C, D or [0])]
+    if dual:
+        _o, d, p = _simplex(-A.T, C.T, B.T, -D)
+        return -_o, d, p
+
+    if A and B:
+        M = Matrix([[A, B], [C, D]])
+    else:
+        assert not A and not B  # no constraints
+        M = Matrix([[C, D]])
+    n = M.cols - 1
+    m = M.rows - 1
+
+    if not all(i.is_Float or i.is_Rational for i in M):
+        # with literal Float and Rational we are guaranteed the
+        # ability of determining whether an expression is 0 or not
+        raise TypeError(filldedent("""
+            Only rationals and floats are allowed in the Simplex method.
+            """))
+
+    # x variables have priority over y variables during Bland's rule
+    # since False < True
+    X = [(False, j) for j in range(n)]
+    Y = [(True, i)  for i in range(m)]
+
+    # Phase 1: find a feasible solution or determine none exist
+    while True:
+        B = M[:-1, -1]
+        A = M[:-1, :-1]
+        if all(B[i] >= 0 for i in range(B.rows)):
+            # We have found a feasible solution
+            break
+
+        # Find k: first row with a negative rightmost entry
+        for k in range(B.rows):
+            if B[k] < 0:
+                break  # use current value of k below
+        else:
+            pass  # XXX is it an error if none was found?
+
+        # Choose pivot column, c
+        piv_cols = [_ for _ in range(A.cols) if A[k, _] < 0]
+        if not piv_cols:
+            raise InfeasibleLPError(filldedent("""
+                The constraint set is empty!"""))
+        c = sorted(piv_cols, key=lambda _: X[_])[0] # Bland's rule
+
+        # Choose pivot row, r
+        piv_rows = [_ for _ in range(A.rows) if A[_, c] > 0 and B[_] > 0]
+        piv_rows.append(k)
+        r = _choose_pivot_row(A, B, piv_rows, c, Y)
+
+        M = _pivot(M, r, c)
+        X[c], Y[r] = Y[r], X[c]
+
+    # Phase 2: from a feasible solution, pivot to optimal
+    while True:
+        B = M[:-1, -1]
+        A = M[:-1, :-1]
+        C = M[-1, :-1]
+
+        # Choose a pivot column, c
+        piv_cols = []
+        piv_cols = [_ for _ in range(n) if C[_] < 0]
+        if not piv_cols:
+            break
+        c = sorted(piv_cols, key=lambda _: X[_])[0] # Bland's rule
```

Comment:
I guess this could be `min` or `max` rather than `sorted`/

Reply:
done

Reply:
Note: I actually struggled with how to say this. I know about the decorated sort but the "decorated max" was a little more ellusive. I got it back to a one-liner, however.